src/test/resources/recipes/MultipleDereferencesRecipes.java

```diff
@@ -59,6 +59,7 @@ public J visitMethodInvocation(J.MethodInvocation elem, ExecutionContext ctx) {
                     if ((matcher = before.matcher(getCursor())).find()) {
                         doAfterVisit(new org.openrewrite.java.ShortenFullyQualifiedTypeReferences().getVisitor());
                         doAfterVisit(new org.openrewrite.java.cleanup.UnnecessaryParenthesesVisitor());
+                        doAfterVisit(new org.openrewrite.staticanalysis.SimplifyBooleanExpression().getVisitor());
```

Comment:
Rather than do a full visit, could we only visit the result or parent?

Reply:
Not quite sure if that would work due to the recursive nature of the recipe, but we can log it as an improvement if you think it should.